renderer/svg.go

```diff
@@ -182,31 +206,30 @@ func sortBreaks(breaks []*models.ChoroplethBreak, asc bool) []*models.Choropleth
 }
 
 // RenderHorizontalKey creates an SVG containing a horizontally-oriented key for the choropleth
-func RenderHorizontalKey(request *models.RenderRequest) string {
+func RenderHorizontalKey(svgRequest *SVGRequest) string {
 
-	geoJSON := getGeoJSON(request)
+	geoJSON := svgRequest.geoJSON
 	if geoJSON == nil {
 		return ""
 	}
-	svg := g2s.New()
-	svg.AppendFeatureCollection(geoJSON)
-	svgWidth, _, _ := getWidthAndHeight(request, svg)
+	request := svgRequest.request
+	svgWidth := svgRequest.width
 
 	keyInfo := getHorizontalKeyInfo(svgWidth, request)
 
 	content := bytes.NewBufferString("")
 	ticks := bytes.NewBufferString("")
 	svgAttributes := fmt.Sprintf(`id="%s-legend-horizontal" class="map_key_horizontal" width="%.f" height="90" viewBox="0 0 %.f 90"`, request.Filename, svgWidth, svgWidth)
 
-	fmt.Fprintf(content, `%s<g id="%s-legend-horizontal-container">`, "\n", request.Filename)
+	fmt.Fprintf(content, `<g id="%s-legend-horizontal-container">`, request.Filename)
 	writeHorizontalKeyTitle(request, svgWidth, content)
-	fmt.Fprintf(content, `%s<g id="%s-legend-horizontal-key" transform="translate(%f, 20)">`, "\n", request.Filename, keyInfo.keyX)
+	fmt.Fprintf(content, `<g id="%s-legend-horizontal-key" transform="translate(%f, 20)">`, request.Filename, keyInfo.keyX)
 	left := 0.0
 	breaks := keyInfo.breaks
 	for i := 0; i < len(breaks); i++ {
 		width := breaks[i].RelativeSize * keyInfo.keyWidth
-		fmt.Fprintf(content, `%s<rect class="keyColour" height="8" width="%f" x="%f" style="stroke-width: 0.5; stroke: black; fill: %s;">`, "\n", width, left, breaks[i].Colour)
-		fmt.Fprintf(content, `</rect>`)
+		fmt.Fprintf(content, `<rect class="keyColour" height="8" width="%f" x="%f" style="stroke-width: 0.5; stroke: black; fill: %s;">`, width, left, breaks[i].Colour)
+		content.WriteString(`</rect>`)
```

Comment:
might be more readable to avoid mixing write styles - e.g. either using `fmt.Fprintf` or `content.Write?` consistently?

Reply:
I've replaced fmr.Fprintf with WriteString as much as possible 0 it does make a difference to performance. I've left it in where we need to format, e.g., numbers.